lumenlearning/lti-middleware
# DL3-3 Switch middleware Deep Linking UI to React

## Description
This PR includes all the work described in https://lumenlearning.atlassian.net/browse/DL3-3.

Incorporates an additional set of UI templates written in React for deep linking requests, this does not affect other type of requests like resources.

The UI displays the new waymaker course picker with some static data until the endpoints are ready to be integrated.

### Motivation and Context
When the user wants to link a Waymaker course through the LMS, via Deep Linking, the LMS sends an LTI 1.3 Deep Linking request, it redirects to the React UI, which contains the new designs of the Waymaker deep linking UI.


### Fixes
[Jira ticket](https://lumenlearning.atlassian.net/browse/DL3-3)

## How Has This Been Tested?
This has been tested in the Unicon's D2L instance using a local development environment, redirecting all the endpoints to the Middleware using Ngrok. 

A proper test plan could be (Not including the LTI deployment stuff):

- As an instructor or admin, go to a course.
- Go to the content tool.
- Add an Existing Activity.
- Select the tool link of the LTI tool.
- Ensure the Waymaker course picker is displayed in the modal.

## Screenshots
![image](https://user-images.githubusercontent.com/8653754/172191238-371f171b-ed29-4218-83ce-de9b2b342104.png)

![image](https://user-images.githubusercontent.com/8653754/172191285-62b0a3d0-daef-4db4-ac6a-5e0db54aecc8.png)
![main](https://user-images.githubusercontent.com/8653754/172349118-0280d06d-a19b-451e-b24b-a451c47468fc.jpg)
![course-info](https://user-images.githubusercontent.com/8653754/172349123-b6f346fe-0850-4c80-97ff-8e6d5c839584.jpg)




## Database changes
No DB changes.

## ⚠️ Deployment instructions ⚠️
No additional instructions.

## New ENV variables
Maybe we must enable deep linking in the Middleware by default.

```
lti13.enableDeepLinking=true
```

---

## Checklist
